### PR TITLE
MultiName FeatureAdded

### LINKED ISSUE #1: Add functionality to run multiple informal names through the pipeline
When this is used by data engineers, they will need to map many concepts in one go and we should accommodate this. Pipeline warmup takes longer than inference, so it will be much faster if the API can take multiple names, warm up the pipeline, then run the names through it.

### PR DESCRIPTION
**Pull request by CodeByKarthik**
Hi James, 

I hope you are doing well. I've made a pull request for handling multiple informal names. The code works well from my side and kindly do the needful. 

Thanks and regards,
Karthik

### CODE REVIEW

File: Carrot-Assistant/install_requirements.sh
Comment (kuraisle) [inline Carrot-Assistant/install_requirements.sh:1]: This needs to be put in your previous pull request so we can close it

Comment (Karthi-DStech) [reply Carrot-Assistant/install_requirements.sh:1]: Ok I'll sort it out.

File: Carrot-Assistant/app.py [hunk 1]
```diff
@@ -34,16 +35,21 @@
...
     """
     This enum holds the names of the different models the assistant can use
     """
-    GPT_3_5_TURBO = "gpt-3.5-turbo-0125",
-    GPT_4 = "gpt-4",
-    LLAMA_2_7B = "llama-2-7B-chat",
-    LLAMA_3_8B = "llama-3-8B",
-    LLAMA_3_70B = "llama-3-70B",
+
+    GPT_3_5_TURBO = ("gpt-3.5-turbo-0125",)
```
Comment (kuraisle) [inline Carrot-Assistant/app.py:48]: Why are these all in tuples now?

File: Carrot-Assistant/app.py [hunk 2]
```diff
@@ -184,6 +201,7 @@ async def run_pipeline(request: PipelineRequest) -> EventSourceResponse:
     """
     return EventSourceResponse(generate_events(request))
 
+
 @app.post("/run_db")
 async def run_db(request: PipelineRequest) -> dict:
```
Comment (kuraisle) [inline Carrot-Assistant/app.py:206]: Can you change the return type?

File: Carrot-Assistant/assistant.py [hunk 3]
```diff
@@ -32,44 +30,48 @@ def run(
...
 
 
 if __name__ == "__main__":
     from options.base_options import BaseOptions
 
     opt = BaseOptions().parse()
-    informal_name = opt.informal_name
-    run(opt=opt, informal_name=informal_name)
+    informal_names = opt.informal_names
+    run(opt=opt, informal_name=informal_names)
```
Comment (kuraisle) [inline Carrot-Assistant/assistant.py:77]: needs to be `informal_names=informal_names`

File: Carrot-Assistant/requirements.txt
Comment (kuraisle) [inline Carrot-Assistant/requirements.txt:1]: As install_requirements.sh

File: Carrot-Assistant/assistant.py (hunk 3 above)
Comment (Karthi-DStech) [reply Carrot-Assistant/assistant.py:77]: I've tested the code after changing into informal_names. I'll make changes

**Review by kuraisle — CHANGES_REQUESTED** [review]
Comments on changes that need to be resolved:
1. Why are the enum values for LLMModel now tuples?
2. Return type in run_db needs to change
3. assistant.py main needs argument updating. This will only come up if it's run from the command line

With these changes I'll be happy to merge, good job

File: Carrot-Assistant/app.py (hunk 1 above)
Comment (Karthi-DStech) [reply Carrot-Assistant/app.py:48]: Thats was another combination I was trying to implement. It's a mistake and I'll change it.

File: Carrot-Assistant/app.py (hunk 2 above)
Comment (Karthi-DStech) [reply Carrot-Assistant/app.py:206]: async def run_db(request: PipelineRequest) -> List[Dict[str, Any]]:

Is this return type correct?

**Review by kuraisle — APPROVED** [review]

**Pull request merged**